### PR TITLE
Refactor phase 1 and fix couple of bugs

### PR DESCRIPTION
**Pull request by toto-dev**

### CODE REVIEW

File: ctools/defragment_sharded_collection.py [hunk 1]
```diff
@@ -387,7 +387,7 @@ def lookahead(iterable):
                 if skip_chunk or not has_more:
                     remain_chunks.append(c)
                     if 'defrag_collection_est_size' not in c:
-                        if not args.dryrun:
+                        if args.dryrun:
```
Comment (toto-dev) [inline ctools/defragment_sharded_collection.py]: I guess this was a typo

File: ctools/defragment_sharded_collection.py [hunk 2]
```diff
@@ -504,6 +496,7 @@ def lookahead(iterable):
                 consecutive_chunks = [c]
                 estimated_size_of_consecutive_chunks = args.phase_1_estimated_chunk_size_kb
                 if not has_more:
+                    await update_chunk_size_estimation(c)
```
Comment (toto-dev) [inline ctools/defragment_sharded_collection.py:499]: Here we were not writing the chunk size estimation for the last chunk, not really a bug, but subsequent loads of `config.chunks` would find a chunk without size estimation

**Review by pierlauro — APPROVED** [review]

File: ctools/defragment_sharded_collection.py (hunk 2 above)
Comment (graetzer) [reply ctools/defragment_sharded_collection.py:499]: ah was  this causing the non-idempotent behaviour?

**Review by graetzer — APPROVED** [review]

File: ctools/defragment_sharded_collection.py (hunk 2 above)
Comment (toto-dev) [reply ctools/defragment_sharded_collection.py:499]: No this is not the main problem, I'll send a subsequent PR with the actual fix for the idempotency ;)

**Pull request merged**